PCbuild/pyproject.props

```diff
@@ -38,9 +38,8 @@
   </PropertyGroup>
   <ItemDefinitionGroup>
     <ClCompile>
-      <AdditionalIncludeDirectories>$(PySourcePath)Include;$(PySourcePath)Include\internal;$(PySourcePath)Include\internal\mimalloc;$(PySourcePath)PC;$(IntDir);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(PySourcePath)Include;$(PySourcePath)Include\internal;$(PySourcePath)Include\internal\mimalloc;$(IntDir);$(PySourcePath)PC;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
       <PreprocessorDefinitions>WIN32;$(_Py3NamePreprocessorDefinition);$(_PlatformPreprocessorDefinition)$(_DebugPreprocessorDefinition)$(_PydPreprocessorDefinition)%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <PreprocessorDefinitions Condition="'$(DisableGil)' == 'true'">Py_GIL_DISABLED=1;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```

Comment:
I think we want to keep this temporarily, so that `Py_GIL_DISABLED` is defined in both `pyconfig.h` and as a command line argument until we can fix up some of the modules. I think this is why the Windows free-threaded builds are failing.

Some modules, like [`_stat.c`](https://github.com/python/cpython/blob/cc7e45cc572dd818412a649970fdee579417701f/Modules/_stat.c#L14-L21) are compiled with `Py_LIMITED_API`, but the `Py_GIL_DISABLED` builds don't work with `Py_GIL_DISABLED`, so there is a check on `Py_GIL_DISABLED`. Unfortunately, `PC/pyconfig.h` both defines `Py_GIL_DISABLED` and depends on `Py_LIMITED_API`, so we may need a bit of refactoring as well as fixing up the modules.

Ideally, `Py_GIL_DISABLED` will eventually work with some form of `Py_LIMITED_API`, but we're not there yet.

Reply:
> There's no good reason to compile our own modules with Py_LIMITED_API...

I generally agree, but other developers want to compile these modules with `Py_LIMITED_API`.

> And we can't assume that anything defined in pyconfig.h is available prior to including pyconfig.h because on non-Windows platforms it won't be.

Yeah, that's what `_stat.c` does. (See the link in the above comment.) But unless we want to expand this PR in scope, it'll be easier to leave this line in temporarily and remove it in a follow up PR that also fixes up `_stat.c` and some other modules on Windows.

Reply:
I'm not going to get into a fight with those guys over it, I'll just get burned out again, and that doesn't help any actual users.

PC/pyconfig.h doesn't depend in any meaningful way on `Py_LIMITED_API`, so we can just remove those checks. When building as part of the stdlib that block is already ignored (and even if it wasn't, it'll be overridden by linker settings in the projects).